src/lib/logger/impl.ml

```diff
@@ -324,13 +324,23 @@ let change_id {null; metadata; id= _} ~id = {null; metadata; id}
 
 let make_message (t : t) ~level ~module_ ~location ~metadata ~message ~event_id
     =
+  let global_metadata' =
+    let m = !global_metadata in
+    let key_cmp (k1, _) (k2, _) = String.compare k1 k2 in
+    match List.find_all_dups m ~compare:key_cmp with
+    | [] ->
+        m
+    | dups ->
+        ("$dup", `List (List.map ~f:(fun (s, _) -> `String s) dups))
```

Comment:
It would be nice to log the metadata for all of the dups here, but that can be a follow-up PR